frontend/javascripts/maintenance_banner.tsx

```diff
@@ -144,3 +148,83 @@ export function MaintenanceBanner() {
 
   return null;
 }
+
+export function UpgradeVersionBanner() {
+  const white = "var(--ant-color-text-primary)";
+  const blue = "var(--ant-color-primary)";
+  const UPGRADE_BANNER_STYLE: React.CSSProperties = {
+    position: "absolute",
+    top: 0,
+    left: 0,
+    height: constants.BANNER_HEIGHT,
+    textAlign: "center",
+    backgroundColor: blue,
+    color: white,
+    fontSize: "medium",
+    minWidth: "fit-content",
+  };
+  const customParseFormat = require("dayjs/plugin/customParseFormat");
+  dayjs.extend(customParseFormat);
+  const currentDate = dayjs();
+
+  const isVersionOutdated = useFetch(
+    async () => {
+      let buildInfo = await getBuildInfo();
```

Comment:
I'd add a `await Utils.sleep(5000)` before this line, so that WK can make more high-priority requests first.